backend/src/main/kotlin/tools/vitruv/vitruvAdapter/vitruv/api/VitruvAdapter.kt

```diff
@@ -2,68 +2,121 @@ package tools.vitruv.vitruvAdapter.vitruv.api
 
 import org.eclipse.emf.ecore.EObject
 import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.emf.ecore.resource.ResourceSet
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl
+import org.eclipse.emf.ecore.xmi.impl.XMIResourceFactoryImpl
 import org.springframework.stereotype.Service
 import tools.vitruv.framework.views.View
 import tools.vitruv.framework.views.changederivation.DefaultStateBasedChangeResolutionStrategy
+import org.eclipse.emf.common.util.URI
+import tools.vitruv.framework.remote.client.VitruvClient
+import tools.vitruv.framework.remote.client.VitruvClientFactory
+import tools.vitruv.framework.remote.client.exception.BadClientResponseException
+import tools.vitruv.framework.views.ViewSelector
+import tools.vitruv.framework.views.ViewType
+import tools.vitruv.vitruvAdapter.vitruv.impl.exception.DisplayViewException
+import java.util.Collections
+import java.util.*
 
 @Service
 class VitruvAdapter {
-    fun getWindows(displayView : DisplayView): Set<String> {
-        val internalSelector = displayView.viewType.createSelector(null)
-        displayView.windowSelector.applySelection(internalSelector)
-        // Now only the names of the windows should be in the selection
-        val view = internalSelector.createView()
-        val windows = mutableSetOf<String>()
-        for (rootObject in view.rootObjects) {
-            // Add all names of windows to windows set
+
+    private var vitruvClient: VitruvClient? = null
+    private var displayViewContainer: DisplayViewContainer? = null
+
+
+    /**
+     * Connects the adapter to the model server.
+     * @param vitruvClient The client to connect to the model server.
+     */
+    fun connectClient(vitruvClient: VitruvClient) {
+        try {
+            vitruvClient.viewTypes
+        } catch (e: BadClientResponseException) {
+            throw IllegalArgumentException("Could not connect to model server.")
```

Comment:
imo its better to throw an own exception like VitruvConnectionFailedConnection so that we can catch the exception and transform it into a http error